dex/src/main/scala/com/wavesplatform/dex/AddressActor.scala

```diff
@@ -442,11 +454,12 @@ class AddressActor(owner: Address,
   }
 
   private def getOrdersToCancel(actualBalance: Map[Asset, Long]): Queue[InsufficientBalanceOrder] = {
+    val inProgress = pendingCancels
     // Now a user can have 100 active transaction maximum - easy to traverse.
     activeOrders.values.toVector
       .sortBy(_.order.timestamp)(Ordering[Long]) // Will cancel newest orders first
       .iterator
-      .filter(_.isLimit)
+      .filter(x => x.isLimit && !inProgress.contains(x.id))
```

Comment:
_Apply_ method of immutable set already returns boolean

Reply:
Don't get what did you mean :) If you are speaking about `inProgress.contains` vs `inProgress`, the first version is more readable.